scheduler/scheduler.go

```diff
@@ -152,3 +157,69 @@ func merge(envs ...map[string]string) map[string]string {
 	}
 	return merged
 }
+
+type Status struct {
+	// A friendly human readable message about the status change.
+	Message string
+}
+
+// String implements the fmt.Stringer interface.
+func (s *Status) String() string {
+	return s.Message
+}
+
+// StatusStream is an interface for publishing status updates while a scheduler
+// is executing.
+type StatusStream interface {
```

Comment:
i like this. i think it helps differentiate `EventStream` as async empire notifications and `StatusStream` as synchronous notifications during the execution of an action.